### ISSUE TITLE
"Pick your data" dialog for "Saved Questions" lists all collections as if they are top level

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#14178]
Maybe not a bug so much as a preference, but with the new collection navigation (v0.38-rc1) it appears inconsistent.

**Describe the bug**
When starting a new question based on a saved question the dialog to select the question presents all the collections in an alphabetical list, you've got to first select a container before you can either search for or select the saved question. We've got quite a few collections and some of the naming is inconsistent which can make it hard to find the exact question you're looking for.

**Logs**

**To Reproduce**
1. Create a nested structure of collections
2. Add a question to each collection
3. Go to "Ask a question" 
4. In the "Pick your data" dropdown expand "Saved Questions"
5. Note that all collections are listed at the same level, there is no nesting shown.

**Expected behavior**
I've always found this a bit jarring, but it feels even more inconsistent now with the direction of collection navigation/visualisation in v0.38.

I expected sub collections to the nested and accessible from the parent collection.

**Screenshots**
Collection setup
![image](https://user-images.githubusercontent.com/4504437/102952556-439e5d00-4534-11eb-83f0-92e633038e73.png)

Saved question selection
![image](https://user-images.githubusercontent.com/4504437/102952606-56189680-4534-11eb-8fad-eee6605b65e4.png)

Bonus, after archiving the only question in "Orders Collection" (the parent collection) the "pick your data" dialog gets a strange empty line in it. This is only on the first time viewing it, if you collapse and expand the empty line goes away. It comes back after a browser refresh.
![image](https://user-images.githubusercontent.com/4504437/102952701-9a0b9b80-4534-11eb-8c3a-ce6a18ebf35a.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-NZ",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "13.0.2+8",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "https://java.oracle.com/",
    "java.version": "13.0.2",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "13.0.2+8",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-12-17",
      "tag": "v0.38.0-rc1",
      "branch": "master",
      "hash": "e8b0bcc"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Obviously not critical. I'd say it's made more apparent by the good work you've done on collection navigation in v0.38. But it would make selecting a saved question simpler if you could use the same collection hierarchy you've already setup within the rest of Metabase

**Comment by flamber** [comment]
The bonus is related to #11571

**Comment by mazameli** [comment]
I'm going to check if this was happening in 0.37.x. We definitely should not be showing all sub-collections at the same hierarchy level. I think this is at least a P2.

**Comment by mazameli** [comment]
Oof, looks like this has been present since at least 0.37.x.

A sub-collection within a collection that has a question:

<img width="261" alt="image" src="https://user-images.githubusercontent.com/2223916/103251448-8b0a5c00-492d-11eb-9a3f-5135c63e6c11.png">

Incorrectly showing all the collections at the top level:

<img width="392" alt="image" src="https://user-images.githubusercontent.com/2223916/103251420-729a4180-492d-11eb-8a31-5cdaab20bd24.png">

**Comment by flamber** [comment]
This was fixed in 0.40 with the complete revamp of the Saved Question picker.
